renku/core/commands/graph.py

```diff
@@ -484,13 +481,32 @@ def as_workflow(
             # Loop through runs and add them as sub processes to parent.
             parent_process.add_subprocess(step.association.plan)
 
-        return self._find_identical_parent_run(parent_process)
+        return self._find_identical_parent_run(run=parent_process, outputs=outputs)
+
+    def _find_identical_parent_run(self, run, outputs):
+        from marshmallow.exceptions import ValidationError
+
+        def workflow_has_identical_subprocesses(workflow_, subprocesses_ids_):
+            wf_subprocesses_ids = [step.process._id for step in workflow_.association.plan.subprocesses]
+            return wf_subprocesses_ids == subprocesses_ids_
```

Comment:
Change to a `set()` instead of a list for comparison or sort the list by index of the `OrderedSubprocess` first. Unfortunately, we can't guarantee the order when deserializing from json-ld (Hence the `OrderedSubprocess` class).

Reply:
Good catch!